### PR TITLE
Update of Licenses; Update of README 

### LINKED ISSUE #4: Rework licensing
After adding new ML-Models, please rework the Licensing (close to [REUSE-Specification](https://reuse.software/))

- [x] Create a folder LICENSES, containing txt-files for every license used in the repo
- [x] For Text-Files: Add SPDX-Headers
- [x] For other files: Create a new file with ending ".license" and add SPDX-Headers
- [x] Create a file LICENSE.md containing a summary of all license-information

### LINKED ISSUE #6: Add Contributors-Section in README.md
- [x] Add section "Contributors" to README.md and list names of all main-contributors with significant contributions

### PR DESCRIPTION
**Pull request by auringonnousu**
Please check before you submit a PR:

- [x] Have you checked to ensure there aren't other open PR for the same update/change?
- [x] Have you read the CONTRIBUTING document?
- [x] Does the commit message follow our guidelines?
- [ ] Have you done your changes in a separate branch?
- [x] Give a descriptive title to you PR.
- [x] Give a description of your changes.
- [x] Other information:

- Integration of license .txt into LICENSES folder
- Add Contributors-Section in README.md

closes #6, closes #4

### CODE REVIEW

**Comment by benjwolff** [comment]
Please add license information for every file (see Issue #4).
 
For ipynb-files you can use *.license-files with SPDX-Data.
For markdown-files use html comments:
```
<!--
SPDX-FileCopyrightText: 2023 Machine-Learning-OER-Collection
SPDX-License-Identifier: CC-BY-4.0
-->
```
Make sure to use "CC-BY-4.0" as identifier instead of "CC-BY 4.0" ([SPDX expects this](https://spdx.org/licenses/)).
Please add the current year to the copyright text.

**Comment by benjwolff** [comment]
There is a license for datasets, but I don't see any datasets in the repository.

**Comment by auringonnousu** [comment]
> There is a license for datasets, but I don't see any datasets in the repository.

currently not able to push data set lfs throwing error, still looking for a way to fix this

**Comment by benjwolff** [comment]
> > There is a license for datasets, but I don't see any datasets in the repository.
> 
> currently not able to push data set lfs throwing error, still looking for a way to fix this

You can push the dataset in a separate PR afterwards if this makes things easier.

File: README.md.license
Comment (benjwolff) [inline README.md.license:1]: The content of this file should be placed on top of the file README.md, not in a separate file. Thus it is commented out as HTML.

Comment (benjwolff) [reply README.md.license:1]: Same goes for all other Markdown files.

Comment (benjwolff) [reply README.md.license:1]: Linting the code with REUSE throws the following summarization:

```
# NICHT VERWENDETE LIZENZEN

Die folgenden Lizenzen werden nicht benutzt:
* CC0-1.0


# FEHLENDE URHEBERRECHTS- UND LIZENZINFORMATIONEN

Die folgenden Dateien haben keine Urheberrechts- und Lizenzinformationen:
* .gitattributes
* .gitignore
* supervised_learning/classification/nearest_neighbors/README.md
* supervised_learning/classification/nearest_neighbors/code/k_nearest_neighbors_classification_penguins.ipynb
* supervised_learning/classification/nearest_neighbors/img/README.md
* supervised_learning/classification/nearest_neighbors/img/accuracy_binary_classification.svg
* supervised_learning/classification/nearest_neighbors/img/confusion_matrix_binary_classification.svg
* supervised_learning/classification/nearest_neighbors/img/confusion_matrix_multi_class.svg
* supervised_learning/classification/nearest_neighbors/img/f1_score.svg
* supervised_learning/classification/nearest_neighbors/img/knn_classification_decision_border_optimal.svg
* supervised_learning/classification/nearest_neighbors/img/knn_classification_decision_border_overfitting.svg
* supervised_learning/classification/nearest_neighbors/img/knn_classification_decision_border_underfitting.svg
* supervised_learning/classification/nearest_neighbors/img/knn_classification_optimal_k_value.svg
* supervised_learning/classification/nearest_neighbors/img/precision_binary_classification.svg
* supervised_learning/classification/nearest_neighbors/img/principle_classification.svg
* supervised_learning/classification/nearest_neighbors/img/principle_knn_classification.svg
* supervised_learning/classification/nearest_neighbors/img/principle_linear_regression.svg
* supervised_learning/classification/nearest_neighbors/img/recall_binary_classification.svg
* supervised_learning/classification/nearest_neighbors/img/type_errors_binary_classification.svg
* supervised_learning/classification/nearest_neighbors/text/accuracy.md
* supervised_learning/classification/nearest_neighbors/text/classification_and_regression.md
* supervised_learning/classification/nearest_neighbors/text/confusion_matrix_binary_classification.md
* supervised_learning/classification/nearest_neighbors/text/confusion_matrix_multi_class.md
* supervised_learning/classification/nearest_neighbors/text/error_types.md
* supervised_learning/classification/nearest_neighbors/text/f1_score.md
* supervised_learning/classification/nearest_neighbors/text/knn_algorithm.md
* supervised_learning/classification/nearest_neighbors/text/optimal_k_value.md
* supervised_learning/classification/nearest_neighbors/text/overfitting_underfitting.md
* supervised_learning/classification/nearest_neighbors/text/precision.md
* supervised_learning/classification/nearest_neighbors/text/recall.md
* supervised_learning/regression/linear_models/linear_regression/README.md
* supervised_learning/regression/linear_models/linear_regression/code/linear_regression.ipynb
* supervised_learning/regression/linear_models/linear_regression/img/README.md
* supervised_learning/regression/linear_models/linear_regression/img/R^2_for_0.5.svg
* supervised_learning/regression/linear_models/linear_regression/img/R^2_for_0.svg
* supervised_learning/regression/linear_models/linear_regression/img/R^2_for_1.svg
* supervised_learning/regression/linear_models/linear_regression/img/R^2_non_linear_distribution.svg
* supervised_learning/regression/linear_models/linear_regression/img/best_fit_line.svg
* supervised_learning/regression/linear_models/linear_regression/img/linear_equation.svg
* supervised_learning/regression/linear_models/linear_regression/text/coefficient_of_determination.md
* supervised_learning/regression/linear_models/linear_regression/text/regression.md


# ZUSAMMENFASSUNG

* Falsche Lizenzen:
* Veraltete Lizenzen:
* Lizenzen ohne Dateiendung:
* Fehlende Lizenzen:
* Unbenutzte Lizenzen: CC0-1.0
* Verwendete Lizenzen: CC-BY-4.0
* Lesefehler: 0
* Dateien mit Urheberrechtsinformationen: 32 / 73
* Dateien mit Lizenzinformationen: 32 / 73

Leider ist Ihr Projekt nicht konform mit Version 3.0 der REUSE-Spezifikation :-(
```
How to fix?:
- Please write SPDX as comment in the header of text-files (md-files and git-settings)
- Add file with ending *.license for all images, containing SPDX

Comment (benjwolff) [reply README.md.license:1]: For .gitignore and .gitattributes CC0-1.0 is the right license (no attribution required).

Comment (benjwolff) [reply README.md.license:1]: Github templates for Issue and PR should also be CC0.

**Comment by benjwolff** [comment]
> For ipynb-files you can use *.license-files with SPDX-Data. For markdown-files use html comments:

In the last commit you have the SPDX inside the ipynb-file. This will not work well. Remove the SPDX again and make a .license file for ipynb notebooks.

**Pull request merged**